tools/project_template/ModuleTemplate/dllmain.cpp

```diff
@@ -114,168 +123,192 @@ class $safeprojectname$ : public PowertoyModuleIface {
+            /*
       if (values.is_string_value(L"string_text_1")) {
         g_settings.string_prop = values.get_string_value(L"string_text_1");
       }
       */
 
-      // Update a color property.
-      /*
+            // Update a color property.
+            /*
       if (values.is_string_value(L"color_picker_1")) {
         g_settings.color_prop = values.get_string_value(L"color_picker_1");
       }
       */
 
-      // If you don't need to do any custom processing of the settings, proceed
-      // to persists the values calling:
-      values.save_to_settings_file();
-      // Otherwise call a custom function to process the settings before saving them to disk:
-      // save_settings();
+            // If you don't need to do any custom processing of the settings, proceed
+            // to persists the values calling:
+            values.save_to_settings_file();
+            // Otherwise call a custom function to process the settings before saving them to disk:
+            // save_settings();
+        }
+        catch (std::exception&)
+        {
+            // Improper JSON.
+        }
     }
-    catch (std::exception& ex) {
-      // Improper JSON.
+
+    // Enable the powertoy
+    virtual void enable()
+    {
+        m_enabled = true;
     }
-  }
 
-  // Enable the powertoy
-  virtual void enable() {
-    m_enabled = true;
-  }
+    // Disable the powertoy
+    virtual void disable()
+    {
+        m_enabled = false;
+    }
 
-  // Disable the powertoy
-  virtual void disable() {
-    m_enabled = false;
-  }
+    // Returns if the powertoys is enabled
+    virtual bool is_enabled() override
+    {
+        return m_enabled;
+    }
 
-  // Returns if the powertoys is enabled
-  virtual bool is_enabled() override {
-    return m_enabled;
-  }
+    // Handle incoming event, data is event-specific
+    virtual intptr_t signal_event(const wchar_t* name, intptr_t data) override
+    {
+        if (wcscmp(name, ll_keyboard) == 0)
+        {
+            auto& event = *(reinterpret_cast<LowlevelKeyboardEvent*>(data));
+            // Return 1 if the keypress is to be suppressed (not forwarded to Windows),
+            // otherwise return 0.
+            return 0;
+        }
+        else if (wcscmp(name, win_hook_event) == 0)
+        {
+            auto& event = *(reinterpret_cast<WinHookEvent*>(data));
+            // Return value is ignored
+            return 0;
+        }
+        return 0;
+    }
 
-  // Handle incoming event, data is event-specific
-  virtual intptr_t signal_event(const wchar_t* name, intptr_t data)  override {
-    if (wcscmp(name, ll_keyboard) == 0) {
-      auto& event = *(reinterpret_cast<LowlevelKeyboardEvent*>(data));
-      // Return 1 if the keypress is to be suppressed (not forwarded to Windows),
-      // otherwise return 0.
-      return 0;
+    // Inherited via PowertoyModuleIface
+    virtual void register_system_menu_helper(PowertoySystemMenuIface* helper) override
```

Comment:
Let's add a comment that `register_system_menu_helper` and `signal_system_menu_action` are part of an experimental features not fully supported yet.